src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1479,6 +1545,9 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node, format
                 impl_candidate = impl_types::ocl;
         }
 
+        if (!deps_for_fully_connected_opt(node, 3)) {
+            impl_candidate = impl_types::ocl;
+        }
```

Comment:
Could you explain why this is necessary? I think this is a WA and should be commented here.

Reply:
Only clDNN reduce returns bf axis oriented output tensor which is used by fully-connected.
Currently, different output size reduce by generation of bf axis oriented uses clDNN by preferred_impl_type.

Reply:
So, remove this logic